src/initialize_dae.jl

```diff
@@ -36,36 +40,52 @@ end
 
 function _initialize_dae!(integrator, prob::ODEProblem,
                           alg::DefaultInit, x::Val{true})
+    isad = SciMLBase.forwarddiffs_model(integrator.alg)
+    nlsolve = NewtonRaphson(autodiff=isad)
     _initialize_dae!(integrator, prob,
-                     BrownFullBasicInit(integrator.opts.abstol), x)
+                     BrownFullBasicInit(integrator.opts.abstol,nlsolve), x)
 end
 
 function _initialize_dae!(integrator, prob::ODEProblem,
                           alg::DefaultInit, x::Val{false})
+    isad = SciMLBase.forwarddiffs_model(integrator.alg)
+    if integrator.u isa Number || integrator.u isa StaticArray
+        nlsolve = SimpleNewtonRaphson(autodiff=isad)
+    else
+        nlsolve = NewtonRaphson(autodiff=isad)
+    end
```

Comment:
this functionality is duplicated. Should it be a method?

Reply:
Chris said "for the oop ones, we want to defualt to SimpleNonlinearSolve when scalar or staticarray so it's fully non-allocating, then the standard NonlinearSolve otherwise"

Reply:
agreed. I just mean that these 5 lines or so are duplicated in about 3 places so I was wondering if it should be pulled out into a function